### DISCUSSION TITLE
React query/TanStack query: should current user be added in the key?

### DISCUSSION THREAD

**Discussion by user72356** [TanStack/query#4345] [Q&A]
I'm just starting with TanStack query and I don't know whether the currently logged in user should be added to the query key of some of my useQuery hook calls?

I use an auth mechanism that spans across the client/server and I do not need to specify the user as a parameter to my API calls. So when my API receives a "getTodos" with no parameters, it knows who the user is and only returns the todo items for this user. I believe this is pretty standard.

Since TanStack query caches the requests based on the key, isn't there a risk that if user1 logs out and user2 logs in from the same browser, the cached data from user1 could be returned to user2?

What is best practice for determining the query key, when the currently logged in user is implied and not used in the query?

**Comment by TkDodo** [comment]
> So when my API receives a "getTodos" with no parameters, it knows who the user is and only returns the todo items for this user.

seems like you have a stateful session in the backend.

you can clear the cache (`queryClient.clear()`) when the user logs out

**Reply by user72356** [reply]
It's actually not a stateful session, just a serverless function that uses JWT for authentication (Supabase auth)
